src/lib/notificationWorker.js

```diff
@@ -31,25 +43,10 @@ class NotificationWorker {
     const affectedAccounts = _([transfer.debits, transfer.credits])
       .flatten().pluck('account').map((account) => this.uri.make('account', account)).value()
     affectedAccounts.push('*')
-
-    let subscriptions = yield this.Subscription.findAll({
-      where: {
-        $and: [{
-          $or: [{
-            event: 'transfer.update'
-          }, {
-            event: 'transfer.*'
-          }, {
-            event: '*'
-          }]
-        }, {
-          subject: {
-            $in: affectedAccounts
-          }
-        }]
-      },
-      transaction
-    })
+    let subscriptions = yield transaction.from('subscriptions')
```

Comment:
Knex has two ways of using transaction.  http://knexjs.org/#Transactions

1.    As the "promise aware" knex connection
 2.  As an object passed into a query with and eventually call commit or rollback.

Method (2) didn't work for me with a generator (see: https://github.com/tgriesser/knex/issues/811), so I exclusively used (1) in Knex.  You are using both methods ... Has (2) been tested thoroughly (e.g. take a transaction, do some insert or update operation, throw an exception and see if the transaction is rolled back)?